### ISSUE TITLE
[Pytorch 1.10.1] fail to install using pip and python 3.10.1

### ISSUE THREAD

**Issue by ntimeu** [pytorch/pytorch#71632]
### 🐛 Describe the bug

Hello,

I am trying to install Pytorch stable(1.10.1) using the pip package manager.

My config is :

- archlinux (latest update)
- archlinux python (version 3.10.1)
- pip version 21.3.1

I am trying to install the version provided by the get-started page on Pytorch's website, using pip and with a cpu setup (no GPU), and I get the following error :

```
$ .pve/bin/pip --verbose install torch==1.10.1+cpu torchvision==0.11.2+cpu torchaudio==0.10.1+cpu -f https://download.pytorch.org/whl/cpu/torch_stable.html
Using pip 21.3.1 from /home/user/.pve/lib/python3.10/site-packages/pip (python 3.10)
Looking in links: https://download.pytorch.org/whl/cpu/torch_stable.html
ERROR: Could not find a version that satisfies the requirement torch==1.10.1+cpu (from versions: none)
ERROR: No matching distribution found for torch==1.10.1+cpu
```

However, when selecting the nightly version, everything goes smoothly :

```
$ .pve/bin/pip --verbose install --pre torch torchvision -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html
Using pip 21.3.1 from /home/user/.pve/lib/python3.10/site-packages/pip (python 3.10)
Looking in links: https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html
Collecting torch
  Using cached https://download.pytorch.org/whl/nightly/cpu/torch-1.11.0.dev20220121%2Bcpu-cp310-cp310-linux_x86_64.whl (170.7 MB)
Collecting torchvision
  Using cached torchvision-0.2.2.post3-py2.py3-none-any.whl (64 kB)
Collecting typing-extensions
  Using cached typing_extensions-4.0.1-py3-none-any.whl (22 kB)
Collecting six
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting pillow>=4.1.1
  Using cached Pillow-9.0.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.3 MB)
Collecting numpy
  Using cached numpy-1.22.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (16.8 MB)
Installing collected packages: typing-extensions, torch, six, pillow, numpy, torchvision
  changing mode of /home/user/.pve/bin/convert-caffe2-to-onnx to 755
  changing mode of /home/user/.pve/bin/convert-onnx-to-caffe2 to 755
  changing mode of /home/user/.pve/bin/torchrun to 755
  changing mode of /home/user/.pve/bin/f2py to 755
  changing mode of /home/user/.pve/bin/f2py3 to 755
  changing mode of /home/user/.pve/bin/f2py3.10 to 755
Successfully installed numpy-1.22.1 pillow-9.0.0 six-1.16.0 torch-1.11.0.dev20220121+cpu torchvision-0.2.2.post3 typing-extensions-4.0.1
```

Also, in nightly mode torchaudio fails to install :

```
$ .pve/bin/pip --verbose install --pre torch torchvision torchaudio -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html
Using pip 21.3.1 from /home/user/.pve/lib/python3.10/site-packages/pip (python 3.10)
Looking in links: https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html
Requirement already satisfied: torch in ./.pve/lib/python3.10/site-packages (1.11.0.dev20220121+cpu)
Requirement already satisfied: torchvision in ./.pve/lib/python3.10/site-packages (0.2.2.post3)
ERROR: Could not find a version that satisfies the requirement torchaudio (from versions: none)
ERROR: No matching distribution found for torchaudio
```

### Versions

(with nightly installed).

```
$ .pve/bin/python collect_env.py
Collecting environment information...
PyTorch version: 1.11.0.dev20220121+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Arch Linux (x86_64)
GCC version: (GCC) 11.1.0
Clang version: 13.0.0
CMake version: version 3.22.1
Libc version: glibc-2.33

Python version: 3.10.1 (main, Dec 18 2021, 23:53:45) [GCC 11.1.0] (64-bit runtime)
Python platform: Linux-5.16.1-arch1-1-x86_64-with-glibc2.33
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.22.1
[pip3] torch==1.11.0.dev20220121+cpu
[pip3] torchvision==0.2.2.post3
[conda] Could not collect
```

cc @ezyang @seemethere @malfet

**Comment by gchanan** [comment]
I don't believe python 3.10 is supported in a release yet.

**Comment by seemethere** [comment]
Yup we have no plan to retroactively support python 3.10 for 1.10.X

Python 3.10 binaries will be available for 1.11
